src/core/server/http/http_config.ts

```diff
@@ -166,3 +174,7 @@ export class HttpConfig {
     this.xsrf = rawHttpConfig.xsrf;
   }
 }
+
+const convertHeader = (entry: any): string => {
+  return typeof entry === 'object' ? JSON.stringify(entry) : String(entry);
+};
```

Comment:
To get exactly the previous behavior back, while keeping the headers conversion logic inside core instead of delegating it to `hapi` as it was done in legacy (where any type of value were sent as headers, letting hapi handle the conversion), I had to use this mixed `JSON.stringify` + `String(of)` approach.

Tell me if this can be improved.

Reply:
Makes sense to avoid breaking the previous behavior.